### PR TITLE
BugFix: check overtime

### PR DESCRIPTION
**Pull request by dranasinghe**
Bugfix: T3 timeout module had a bug. now the datatime objects convert the time to seconds and takes the difference.

### CODE REVIEW

**Comment by alongd** [comment]
Instead of storing seconds since the epoch, perhaps store a `datetime` object? It's then easy to subtract these objects to get a `timedelta` [object](https://docs.python.org/2/library/datetime.html#timedelta-objects).

**Comment by kspieks** [comment]
@alongd So you're saying to convert `time.time()` to be a `datetime` object, so that line 1241 in `check_overtime` in `main.py` is subtracting `datetime` objects? For example, change `delta = time.time() - self.t0` to be `delta = dt.datetime.fromtimestamp(time.time()) - self.t0` ?

**Comment by alongd** [comment]
Just as a suggestion, we could store times as `datetime.datetime`, and then compare two `datetime.timedelta`'s. If you'd like, you could play around with the code below.
```python
import datetime
import time  # importing just for the sleep() function

t0 = datetime.datetime.now()  # a datetime object
time.sleep(3)
t1 = datetime.datetime.now()  # a datetime object

delta = t1 - t0  # a timedelta object
max_time = datetime.timedelta(1, 2, 3)  # hrs, sec, min  # a timedelta object

print(delta)

if delta > max_time: # comparing timedelta's
    print('over time')
else:
    print('OK')
```

**Comment by kspieks** [comment]
Yea I was just about to say the same thing. I'll probably update the definition of the t0 attribute on line 135 so that it stores a datetime object rather than just converting it in `check_overtime`

**Comment by kspieks** [comment]
@alongd Let me know if the pushed changes are what you meant. It now stores everything as a datetime object and no longer requires changing the unittest. Thank you Duminda for catching the issue and thank you Alon for the feedback :)

**Comment by kspieks** [comment]
Upon trying to fix the failed tests, I'm realizing that both ARC and RMG seem to operate with `time.time()`, which is the timestamp in seconds. So even if we make T3 consistently use datetime objects, T3's functional tests fail due to `TypeError: unsupported operand type(s) for -: 'float' and 'datetime.datetime'`. Because of this, I'd vote to use Duminda's original change, which is to have T3 work with timestamps. Let me know if you have different thoughts.

**Comment by alongd** [comment]
I added a temporary commit. Let me know if it looks reasonable, and I'll organize it

**Comment by kspieks** [comment]
Thanks. The new `time_lapse` function looks good

**Comment by dranasinghe** [comment]
Is it possible to merge the branch to the master today?

**Comment by kspieks** [comment]
Yea I'll review it now

**Comment by dranasinghe** [comment]
Thank you!

File: t3/main.py [hunk 1]
```diff
@@ -107,7 +106,7 @@ class T3(object):
         rmg (dict): RMG directives.
         qm (dict, optional): QM directive.
         verbose (int): The logging level.
-        t0 (float): Initial time when the project was spawned.
+        t0 (float): Initial time when the project was spawned, stored as a datetime object.
```
Comment (kspieks) [inline t3/main.py:109]: Super minor but I had forgotten to update this docstring from float to datetime.datetime. I'll push a minor fix.

**Review by kspieks — COMMENTED** [review]
Thanks for organizing the commits Alon. Looks good

**Review by kspieks — APPROVED** [review]

**Pull request merged**